### PR TITLE
Implement file downloads 

### PR DESCRIPTION
**Pull request by IgnacioPro**
This PR introduces the download of files for sosreports.
    The user can specify amount of days, months or years from
    current date to fetch data. Use --days --months --years

### CODE REVIEW

File: cmd/adhoc/cmd/download.go
Comment (cescgina) [inline cmd/adhoc/cmd/download.go]: Minor thing, but shouldn't this 3 be 2? Since the minimum (and default) amount of days is 2

File: pkg/bucketaws/setup.go
Comment (cescgina) [inline pkg/bucketaws/setup.go]: I think that the return type of `ConfigRead` should be adjusted, since it no longer returns any error, it either succeeds and returns nil, or fails and exits

File: pkg/bucketaws/setup.go
Comment (cescgina) [inline pkg/bucketaws/setup.go:98]: Same issue with `ConfigRead` IMO

File: cmd/adhoc/cmd/download.go
Comment (IgnacioPro) [reply cmd/adhoc/cmd/download.go]: No, because Syncron actually counts today too, so it's for today, yersterday and day before yesterday.

File: pkg/bucketaws/setup.go
Comment (IgnacioPro) [reply pkg/bucketaws/setup.go]: Good point, I will adjust it

File: pkg/bucketaws/setup.go
Comment (IgnacioPro) [reply pkg/bucketaws/setup.go:98]: On this one I am actually returning an error in case of failure.

Comment (cescgina) [reply pkg/bucketaws/setup.go:98]: Right, I missed that, sorry!

Comment (IgnacioPro) [reply pkg/bucketaws/setup.go:98]: No problem, thanks for the review Joan.

File: cmd/adhoc/cmd/download.go
Comment (jsanemet) [reply cmd/adhoc/cmd/download.go]: This is confusing. Does setting '--days' to 2 mean today, yesterday and the day before?

File: cmd/adhoc/cmd/download.go
Comment (jsanemet) [inline cmd/adhoc/cmd/download.go]: Why the difference in variable naming style?

File: cmd/adhoc/cmd/download.go
Comment (jsanemet) [inline cmd/adhoc/cmd/download.go]: I have my doubts when it comes to logging, but I think it is best to join all messages related to an error into a single one. Otherwise it looks like you have more problems than you do.
`Error: No data available. Try again with...`
vs
```
Error: No data available.
Error: Try again with...
```

File: pkg/bucketaws/setup.go
Comment (jsanemet) [inline pkg/bucketaws/setup.go]: Documentation in Go goes on top of the function I believe. Maybe not.

Comment (jsanemet) [inline pkg/bucketaws/setup.go]: You could be a little bit more specific about this function. A rough example:
```
Builds dates in S3 format from the given time to today jumping one day at a time in that direction.
```

File: pkg/bucketaws/setup.go
Comment (jsanemet) [inline pkg/bucketaws/setup.go]: You can do both with a fatal message instead.

File: pkg/bucketaws/setup.go
Comment (jsanemet) [inline pkg/bucketaws/setup.go]: In this case you will want to print into stderr instead of stdout. Must be a way to specify so.

File: pkg/bucketaws/setup.go
Comment (jsanemet) [inline pkg/bucketaws/setup.go:98]: Instead of passing a complicated to generate argument like 'dates', why don't you just pass the time here and process it inside instead?

File: pkg/bucketaws/setup.go
Comment (jsanemet) [inline pkg/bucketaws/setup.go]: I am not fond of crashing the application against an error. What if as a caller I wanted to try again with another configuration? It is unexpected behavior that makes things harder to work with.

**Review by jsanemet — APPROVED** [review]
Overall, nothing that would justify rejecting the merge. The thing that troubles me the most is how the '--days' argument works.

File: cmd/adhoc/cmd/download.go
Comment (IgnacioPro) [reply cmd/adhoc/cmd/download.go]: Yes, as the tool suggests, you are asking syncron to download from the past "x" days, so you are asking for the past 2 days (yesterday and day before yesterday) as well as today.

File: cmd/adhoc/cmd/download.go
Comment (IgnacioPro) [reply cmd/adhoc/cmd/download.go]: Yes, like mentioned on another comment, it is downloading the previous two days and current day.
Will fix this inconsistent variable naming.

File: cmd/adhoc/cmd/download.go
Comment (IgnacioPro) [reply cmd/adhoc/cmd/download.go]: Very good point, fixed.

File: pkg/bucketaws/setup.go
Comment (IgnacioPro) [reply pkg/bucketaws/setup.go]: You are right, fixed.

File: pkg/bucketaws/setup.go
Comment (IgnacioPro) [reply pkg/bucketaws/setup.go]: Is using logrus.Fatal also applicable in this situation?

File: cmd/adhoc/cmd/download.go
Comment (jsanemet) [reply cmd/adhoc/cmd/download.go]: But it says "from the last X days", which for 2 is today and yesterday. I don't know, I saw it easier to understand before.

File: pkg/bucketaws/setup.go
Comment (IgnacioPro) [reply pkg/bucketaws/setup.go:98]: Because I wanted to have a separate function which made things more accesible and readable.

File: pkg/bucketaws/setup.go
Comment (jsanemet) [reply pkg/bucketaws/setup.go]: I think you could.

File: pkg/bucketaws/setup.go
Comment (IgnacioPro) [reply pkg/bucketaws/setup.go]: This was my way of interrupting the execution, as it was throwing the error message, but the application was continuing.
Any suggestions for a better implementation?
I guess going with Fatal again...

File: cmd/adhoc/cmd/download.go
Comment (IgnacioPro) [reply cmd/adhoc/cmd/download.go]: My logic is telling me last 2 days means those days are in the past, and then today, the present.
I have no strong opinion on this, I can make the change and move on.

File: pkg/bucketaws/setup.go
Comment (cescgina) [reply pkg/bucketaws/setup.go]: I think that what @jsanemet meant in this comment is not so much about `Fatal` or `Error`+`Exit` (according to the documentation `Fatal` does internally already call `Exit`, so it's essentially the same), but rather that maybe the function should return the error and then it should be the caller who should decide whether to throw or not. I agree with him, I would not expect a call to this function to interrupt the program, but it depends on how it's intended to be called.

**Review by cescgina — APPROVED** [review]
I think it can be merged, I just left a comment regarding the use of `Fatal` and returning errors, but nothing that should block the PR IMO

**Pull request merged**
